### ISSUE TITLE
When I conveter a tensorflow model  to TFlite，I found that only some parameters are quantized as int8 , not all parameters，why?

### ISSUE THREAD

**Issue by LNCC99** [tensorflow/tensorflow#76422]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 22.04
- TensorFlow installation (pip package or built from source): Tensorflow 2.4
-            tensorboard               2.11.2                   pypi_0    pypi
              tensorboard-data-server   0.6.1                    pypi_0    pypi
              tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
               tensorflow                2.4.0                    pypi_0    pypi
               tensorflow-estimator      2.4.0                    pypi_0    pypi
              tensorflow-model-optimization 0.7.3                    pypi_0    pypi
              termcolor                 1.1.0                    pypi_0    pypi
                tflite-runtime            2.7.0                    pypi_0    pypi
- TensorFlow library (version, if pip package or github SHA, if built from source): Tensorflow 2.4

### 2. Code
After conveter to .tflite，then I got the parameters。I found only the Dense layer's weight is int8，the parameter bias and Conv2D parameters are also float32。Accrodong to the conveter instruction， I think  all the parameters are int8 ,is that right? Thank you!
The code is define as follows :
  mnist = tf.keras.datasets.mnist
(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

# Normalize the input image so that each pixel value is between 0 to 1.
train_images = train_images.astype(np.float32) / 255.0
test_images = test_images.astype(np.float32) / 255.0

# Define the model architecture
model = tf.keras.Sequential([
  tf.keras.layers.InputLayer(input_shape=(28, 28)),
  tf.keras.layers.Reshape(target_shape=(28, 28, 1)),
  tf.keras.layers.Conv2D(filters=12, kernel_size=(3, 3), activation='relu'),
  tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),
  tf.keras.layers.Flatten(), #
  tf.keras.layers.Dense(10)  # 
])
model.summary()
# Train the digit classification model
model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(
                  from_logits=True),
              metrics=['accuracy'])
model.fit(
  train_images,
  train_labels,
  epochs=5,
  validation_data=(test_images, test_labels)
)

## then converter to tflite
converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_model_quant = converter.convert()

**Comment by gaikwadrahul8** [comment]
Hi, @LNCC99

I apologize for the delayed response, I observed the same behavior as far I know it’s typical for certain parameters like biases to remain float32 after quantization and this is a designed behavior in TFLite

This is what you seem to be using with `tf.lite.Optimize.DEFAULT`. It generally quantizes weights of layers like `Dense` and `Conv2D` to int8. However, biases are often kept as `float32` to maintain precision especially since they can affect the output significantly. Biases in layers like `Dense` and `Conv2D` are usually kept as `float32`. This is a common practice to ensure that the models accuracy is not compromised.
 
If you want to enforce that all weights and biases are quantized to `int8`, you'll have to do something like below please refer this [gist-file](https://colab.sandbox.google.com/gist/gaikwadrahul8/c35891f5e9f328cec151bd65e83492a8/test-76422.ipynb) but you'll observe certain amount of drop in the model accuracy

Please refer our official documentation for [Model optimization](https://ai.google.dev/edge/litert/models/model_optimization) because we do have different quantization schemes you might want to use for your model.

```
converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]

# Adding a representative dataset for better quantization
def representative_dataset():
    for i in range(100):
        yield [train_images[i:i+1].astype(np.float32)]  # Make sure to use float32

converter.representative_dataset = representative_dataset
converter.target_spec.supported_types = [tf.int8]  # Ensure int8 quantization

tflite_model_quant = converter.convert()
```
 
I have missed something here please let me know.

Thank you for your cooperation and patience.

**Comment by LNCC99** [comment]
Hi, gaikwadrahul8
    Thank you very much for your answer! Excepting the bias ,the weights of Conv2D are also float32 .I think the weights of Conv2D should be quantized as int8, not float32。Why the weights of Conv2D are also float32?
    Thank you very much!

**Comment by gaikwadrahul8** [comment]
Hi, @LNCC99

I apologize for the delayed response, just to confirm, are you printing above results with quantized model with my notebook to avoid confusion? if possible could you please try with https://netron.app/ by loading the quantized model and see is it still showing the weights and biases of `Conv2D` layer float32 to cross verify

Thank you for your cooperation and patience.

**Comment by LNCC99** [comment]
Hi，@gaikwadrahul8
      Yes，the printing is with your notbook。The weights of Conv2D are still float32 in netron.app:
      
![image](https://github.com/user-attachments/assets/6a800634-98c1-4777-a6dd-ffc0890565d9)

**Comment by gaikwadrahul8** [comment]
Hi, @LNCC99

I apologize for the delayed response, please refer this updated [gist-file](https://colab.sandbox.google.com/gist/gaikwadrahul8/ee363d47c98da3cd98266216e0cb0877/tflite-76422.ipynb) and it seems like things are working as expected now I'm getting below output in netron app so please give it try from your end and let me know things are working as expected or not ?

![image](https://github.com/user-attachments/assets/fa8071f6-bab5-4705-a6be-062df07ae04a)

Thank you for your cooperation and patience.

**Comment by LNCC99** [comment]
Hi,@gaikwadrahul8
     Thank you for your answer。In your "gist-file",the quantization is fully-quantized, in this quantization ,the weights and the activate  are all int8,but I mean that using the dynamic range quantization.I want only to quantizate the weights but not contain the activates . In the dynamic range quantization ,why the weights of conv2D are also float32?
